sdk/core/azure_core/src/policies/mod.rs

```diff
@@ -1,10 +1,12 @@
 mod custom_headers_policy;
 mod retry_policies;
+mod storage_bearer_token_policy;
```

Comment:
We should not include `storage` anywhere for this since it is going into Core and will be common for all packages. `bearer_token_policy` is probably fine for the module.

Though I am curious on what this is called in other languages. Isn't it something like TokenCredentialPolicy?

Reply:
https://github.com/Azure/azure-sdk-for-net/blob/ede7138b0150024daa440819cc0008cd51f46bc1/sdk/core/Azure.Core/src/Pipeline/BearerTokenAuthenticationPolicy.cs#L17

.NET uses `BearerTokenAuthenticationPolicy`